Automattic/wp-calypso
# Fix: Calypso Page/Posts screen on Atomic sites reverts to old design

Closes https://github.com/Automattic/wp-calypso/issues/69908
Closes https://github.com/Automattic/wp-calypso/issues/49486

## Proposed Changes

This PR fixes the redirect to `wp-admin` after the users clicks on `All pages` or `All posts` when exiting a post or page editor when a) the default interface is enabled and b) the condition `is_jetpack_site_eligible_for_gutenframe` returns `false` on an Atomic site.

## Testing Instructions



**Before testing an actual issue, we would need to have the following setup**:

* Create an Atomic site (you can use WoA site)
* Install an Elementor plugin on an Atomic site (you can search for it from `Plugins` > `Add New Plugin`: https://wordpress.com/plugins/{yourAtomicSite)
* Activate Elementor plugin 
* Navigate to `Settings > Hosting Configuration`
* Set the `Admin interface style` on an Atomic site to `Default style`
* Navigate to `Plugins > Installed plugins` and disable `WordPress.com Editing Toolkit` plugin on your Atomic site
* Navigate to [Build Calypso Apps (WPCom Plugins) ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WPComPlugins_Build_Plugins/12299310?buildTab=artifacts)and download a .zip file for the `Editing Toolkit` plugin (available under the `Artifacts` tab):

<img width="984" alt="Screenshot 2024-04-12 at 3 48 34 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/2bee173f-82e9-471c-a267-702df22184b2">

* Navigate to  `Plugins` > `Add New Plugin` on your Atomic siteL https://wordpress.com/plugins/{yourAtomicSite)
* Install the .zip file of the [editing-toolkit.zip](https://teamcity.a8c.com/repository/download/calypso_calypso_WPComPlugins_Build_Plugins/12299310:id/editing-toolkit.zip) that you just downloaded

**Testing an actual issue**:

* Once the steps above are completed, you are ready to test the issue
* Navigate to `Posts` page on your Atomic site: `https://wordpress.com/posts/{yourAtomicSite}`
* Open a post to edit (it does not matter which one, could be any post)
* Click on the WordPress icon in the top left corner to open sidebar navigate
* Click on `All Posts` button
* Confirm that you are brought back to Calypso URL and you see the default interface and not `wp-admin` interface

<img width="1247" alt="Screenshot 2024-04-12 at 3 55 08 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/15ba2941-7bcb-48a1-81e8-46623726fda9">

* Repeat the same steps for the `Pages`: `https://wordpress.com/pages/{yourAtomicSite}`
* Confirm that you are succesfully brought to Calypso URL 

**Testing that other flows work as expected **:

**Case 1: Disable Elementor + Default admin interface**:

* Deactivate Elementor plugin
* Navigate to `Settings > Hosting Configuration`
* Set the `Admin interface style` on an Atomic site to `Default style`
* Navigate to `Posts` page on your Atomic site: `https://wordpress.com/posts/{yourAtomicSite}`
* Open a post to edit (it does not matter which one, could be any post)
* Click on the WordPress icon in the top left corner to open sidebar navigate
* Click on `All Posts` button
* Confirm that you are brought back to Calypso URL and you see the default interface and not `wp-admin` interface

**Case 2: Disable Elementor + Classic admin interface**

* Deactivate Elementor plugin
* Navigate to `Settings > Hosting Configuration`
* Set the `Admin interface style` on an Atomic site to `Classic style`
* Navigate to `Posts` page on your Atomic site (this should now be showing you wp-admin interface)
* Open a post to edit (it does not matter which one, could be any post)
* Click on the WordPress icon in the top left corner to open sidebar navigate
* Click on `All Posts`/`View Posts` button
* Confirm that you are brought back to `wp-admin` interface

**Case 3: Enable Elementor + Classic admin interface**

* Activate Elementor plugin
* Navigate to `Settings > Hosting Configuration`
* Set the `Admin interface style` on an Atomic site to `Classic style`
* Navigate to `Posts` page on your Atomic site (this should now be showing you wp-admin interface)
* Open a post to edit (it does not matter which one, could be any post)
* Click on the WordPress icon in the top left corner to open sidebar navigate
* Click on `All Posts`/`View Posts` button
* Confirm that you are brought back to `wp-admin` interface

## Pre-merge Checklist
